Merge upstream Fusaka readiness

piersy opened the pull request:
Merges to the tip of the upstream main branch.

Closes https://github.com/celo-org/celo-blockchain-planning/issues/1271
Closes https://github.com/celo-org/celo-blockchain-planning/issues/1260

fakedev9999 committed:
chore: support contract verification for fp (#555)

yuwen01 committed:
feat: validate agg (#552)

fakedev9999 committed:
test: improve e2e testing with anvil (#549)

fakedev9999 committed:
ci: add link + spell checker (#564)

* ci: add link + spell checker

* ci: remove recursive submodules checkout

* docs: use different link for spn

0xsharma committed:
feat: Add comprehensive logging for range and aggregation proof requests (#560)

* add : essential logs for range+agg proofs

* chg : if/else into match

---------

Co-authored-by: Yuwen Zhang <7747842+yuwen01@users.noreply.github.com>

fakedev9999 committed:
chore: bump kona (#554)

* chore: bump kona

* refac: avoid allocations when handling input

* refac: avoid alloc + clone

* chore: nits

fakedev9999 committed:
feat: prove task improvements (#558)

* feat: add game prove metrics

* feat: add fast finality proving limit

* chore: nit fixes

fakedev9999 committed:
feat: improve fault proof (#563)

* feat: improve game resolution

* feat(fault-proof): separate bond claiming by mode

* docs: fix typo

* feat: set jemalloc as global allocator

fakedev9999 committed:
feat: update to support celestia indexer changes (#561)

* feat: update to support celestia indexer changes

See https://github.com/celestiaorg/optimism/pull/502.
Now handles batcher that switch between DA layers.

* chore: fix logger init

* chore: bump hana for v4 compatibility

* chore: bump celestia range elf

yuwen01 committed:
chore: pin nightly rust toolchain (#580)

* dummy

* pin nightly version

yuwen01 committed:
feat: audit fix (#575)

* tests pass, fixes in

* proposer is running locally

* fix test, still broken though

* more detailed error

* jank proxy

* wrapper solution instead

* cleanup

* fix silly errors;

* tweak forge test

* docs for dgfProposeL2Output;

* spot=false

* bump elf

* nits + docs

* add audit + pin nightly toolchain

fakedev9999 committed:
chore: unify binary name for celestia docker image (#577)

fakedev9999 committed:
feat: improve op-succinct-lite (#583)

* chore: add features support to deploy-fdg-contracts in justfile

* feat(contracts): add missing getters for public values

fakedev9999 committed:
fix: use end exclusive range (#585)

* fix: end exclusive range

See https://docs.celestia.org/how-to-guides/blobstream-proof-queries.

* refactor(scripts): host method for starting block

* refac: handle when None returned

* change for fetch-l2oo-config as well

fakedev9999 committed:
chore: bump hana (#578)

fakedev9999 committed:
feat: add cycles + prover gas metrics (#579)

* feat: add cycles + prover gas metrics

Also improved logging for better vis and debugging.

* chore: make fmt happy

* chore: apply comments

fakedev9999 committed:
docs: update spn usage (#592)

yuwen01 committed:
docs: make update parameters flow more clear (#590)

* fix

* doc nits

yuwen01 committed:
fix: kzg verification 3.1.0 (#594)

* fix

* refresh gh

yuwen01 committed:
chore: range proof insertion log to debug

0xernesto committed:
docs: L2 node latency + cleanup (#598)

* add dev section to readme

* doc pass

fakedev9999 committed:
chore: bump to op-contracts v3.0.0 (#600)

fakedev9999 committed:
test: fix e2e test (#605)

ctian1 committed:
fix(validity): request deadline (#604)

* fix(validity): request deadline

* fix(fault-proof): request deadline

* fix: add 1T gas limit

The DEFAULT_GAS_LIMIT is 1B.

* fix: add 4h timeout to agg proof request as well

---------

Co-authored-by: fakedev9999 <taehoon@succinct.xyz>

fakedev9999 committed:
feat: add op-celestia-indexer to docker compose (#601)

* docs: add missing CELESTIA_INDEXER_RPC

* feat: add op-celestia-indexer to docker compose

* docs: update docs

* fix healthcheck

* docs: add warning for proof window

* docs: udpate start l1 block and op-celestia-indexer

fakedev9999 committed:
fix(contracts): upgrade, missing proxy in calldata for proxy admin managed instance (not naked proxies) (#606)

* fix(contracts): upgrade, missing proxy in calldata for proxy admin managed instance (not naked proxies)

* chore: forge fmt

* refac(contracts): use abi.encodeCall

---------

Co-authored-by: emiliano-conduitxyz <emiliano@conduit.xyz>

fakedev9999 committed:
feat: support eigenda (#572)

* feat: support eigenda

Bumps sp1, hana, hokulea for kona-client/v1.0.2 with MSRV 1.86.

* ci: install sp1 toolchain

* chore: bump elfs

* fix: write proof to stdin for proof agg

* fix: bump hokulea + hardcoded vkey + defer verification

* fmt

* bump elfs

* feat: docker for eigenda

* refac: extract proof bytes from witness data

* refac: remove executor placeholder

* chore: download srs from proxy image

* docs: add eigenda docs

* chore: bump hokulea

* make ci green

* skip canoe proof generation

* chore: bump hokulea

* bump hokulea to support mock mode for sp1-cc proving

fakedev9999 committed:
fix: reduce RPC concurrency to prevent rate limit errors (#608)

* chore: remove unused code

* fix: reduce RPC concurrency to prevent rate limit errors

fakedev9999 committed:
chore: bump hana to v1.0.0 (#612)

fakedev9999 committed:
chore: add missing eigenda flag (#610)

fakedev9999 committed:
docs: make altda deployments more clear (#615)

leruaa committed:
feat: spawn defense tasks concurrently (#614)

* feat: spawn defense tasks concurrently

* fix: typo

* feat: limit concurrent proving tasks count

* feat: add `count_active_defense_tasks()`

fakedev9999 committed:
fix(fault-proof): filter by configured game type (#616)

* fix(fault-proof): filter by configured game type

* test: add e2e test for game type transition

* fix bindings

* forge fmt

fakedev9999 committed:
feat: add env-configurable proof fulfillment strategies (#618)

* feat: add env-configurable proof fulfillment strategies

* make ci happy

* make ci happy2

fakedev9999 committed:
chore: bump to v3.2.0 (#619)

fakedev9999 committed:
ci: enable release tagging for lite docker images (#620)

fakedev9999 committed:
chore: bump hana (#621)

* chore: bump hana

* chore: bump to v3.2.1.

* chore: bump hana to v1.0.1

leruaa committed:
feat: get execution stats from network when not in mock mode (#622)

fakedev9999 committed:
ci: add cycle count diff ci for celestia range elf (#573)

* ci: add cycle count diff ci for celestia range elf

* chore: increase block selection offset to 2 hours

This is a simple solution for Celestia Blobstream availability

* ci: add elf label

* fix to use finalized l2 block

* fix typo

fakedev9999 committed:
chore: bump hokulea (#609)

* chore: bump hokulea

* ci: add eigenda to elf ci

* ci: add cycle count diff check for eigenda

* ci: fix srs error

* fix copy srs

* fix

* OP_SUCCINCT_MOCK=true for sp1-cc

* fix

* chore: bump eigenda elf (#623)

* bump elfs

* chore: clamp highest provable L2 block

This avoids proving uncommitted Celestia ranges.

fakedev9999 committed:
docs: fix link to not use absolute path (#625)

This prevents 404s when the book is hosted under a sub-path.

fakedev9999 committed:
fix: bump hokulea version and fix only ethda problem (#626)

* bump hokulea version and fix only ethda problem

* name nit

* rename eigenda_blob_provider to eigenda_preimage_provider

* update elf

* restore todo

* chore: bump elf

---------

Co-authored-by: BowenX <bxue@eigenlabs.org>

fakedev9999 committed:
chore: bump hana (#630)

* chore: bump hana to v1.1.0-mocha

* chore: bump to v3.2.2

* chore: bump elfs

fakedev9999 committed:
feat: skip CHALLENGER_WINS game resolution in proposer mode (#632)

Added a condition to skip resolution when the parent game status is CHALLENGER_WINS, since such games will be ultimately resolved as CHALLENGER_WINS. Proposers have no incentive to resolve these games.

Co-authored-by: seolaoh <osa8361@gmail.com>

fakedev9999 committed:
fix(fault-proof): fix proposer with canonical chain tracking (#613)

* fix(fault-proof): fix proposer with canonical chain tracking

* refactor: improve game resolution

* refactor: improve game credit claim

* refactor: improve game defense

* chore: add comments

* chore: rename to update_canonical_head_if_further

* chore: rename to is_XXX for boolean returning func

* refactor: remove early return from `refresh_state`

* fix

* fix

* refactor: refactor fix-proposer PR (#627)

* move run() to the top

* move state init from refresh state to proposer init

* huge refac

* remove is_finalized

* remove update_canonical_head_if_further

* move below new

* switch orders

* compute canonical head

* remove state snapshot

* typo

* refac remove subtree

* fmt

* update fetch_game

* rename to sync_state

* remove is_descendant_of_anchor

* refac remove_subtree

* more refac (#628)

* wip

* remove for claim_bond_for_game

* fix

* clippy

* fix

* fix

* fix

* clippy

* fix

* refac

* fix: drop if parent game is not in cache

* switch orders

* fix is_parent_resolved

* update testing doc

* remove cursor and set_cursor

* improve fetch_game

* improve sync_games

* misspell

* test: add comments to e2e tests

0xernesto committed:
feat: optionally request proofs on spn mainnet (#587)

* bump sp1 sdk

* add support for KMS requester

* update binaries

* use local signer

* use auction strategy

* fix tests and binaries

* fmt

* make timeout 1 hour

* fix cycle/gas limits

* add env vars for more prove params

* whitelist env var in validity proposer

* whitelist for faul-proof

* bump sp1 sdk & elfs

* don't split if cancelled

* sqlx + add auction_timeout env var

* fix proof request cancellation

* fix sqlx

* return early

* fix conditional check

* fix cancel logic

* return early if cancelled

* fix log

* fix sqlx

* add min auction period to requests

* fix: build errors

* feat: determine network mode from fulfillment strategies

* fix: typos

* fix: strip "0x"

* fix: set max price per pgu to 300,000,000

* fix: handle UnspecifiedFulfillmentStrategy

* fix: add handle_cancelled_request()

* fix: configure min auction period with a env var

* fix: cancel requests only if NetworkMode is Mainnet

* chore: add the new env vars in the docs

* feat: refactor network signer creation to a util fn

* fix: feedbacks

* chore: bump SP1 to v5.2.2

* fix: pin SP1 version

* chore: update ELFs

* fix: rebase

* fix: Docker tag

* chore: update ELFs again

* fix: typo

* fix: rebase

---------

Co-authored-by: leruaa <aurelien.catinon@gmail.com>

fakedev9999 committed:
fix: retain canonical head when pruning resolved games (#634)

* fix: retain canonical head when pruning resolved games

* clippy

fakedev9999 committed:
feat: resume fast finality proving after restart (#637)

yuwen01 committed:
feat: range proof gas splitting (#576)

* tests pass, fixes in

* proposer is running locally

* fix test, still broken though

* more detailed error

* jank proxy

* wrapper solution instead

* cleanup

* fix silly errors;

* tweak forge test

* docs for dgfProposeL2Output;

* works locally

* cleanup

* forge fmt

* chore: update ELFs

* feat: add a range limit when computing ranges by gas

* fix: fmt

* feat: rename gas_limit to evm_gas_limit

* fix: revert RANGE_PROOF_INTERVAL default value

* fix: rename RANGE_PROOF_GAS_LIMIT to RANGE_PROOF_EVM_GAS_LIMIT

* feat: replace tuples by Range type

* chore: update comment

* chore: add `single_small_block` test

* fix: comment

* fix: add a warning if no ranges were inserted in the database

* feat: merge disjoint ranges if possible before splitting

* fix: perf feedbacks

---------

Co-authored-by: leruaa <aurelien.catinon@gmail.com>

fakedev9999 committed:
chore: bump kona for Fusaka readiness (#633)

* kona bump wip

* patch kona

* compiles

* fmt

* revert

* chore(kona): bump version

* finally works

* chore: missing changes from merge commit

* bump to tag

* bump elfs

* fix l1 config path

* feat: support load l1 chain config from fs

* fix: cycle count on a range from 1 week ago

* revert: cycle count on a range from 1 week ago

This reverts commit ea93264558adcd43909fa42d73ca7ec3c398c7c2.

* chore: bump hana (#640)

* bump hana

* chore: bump version

* bump elfs

* bump hokulea

* fix

* test: mine blocks every 1 second

* switch e2e test ci to use mainnet endpoints

* revert

* fix

* fix

---------

Co-authored-by: theochap <theodore.chap@gmail.com>
Co-authored-by: leruaa <aurelien.catinon@gmail.com>

fakedev9999 committed:
fix: add timeouts to network prover calls to prevent deadlock (#639)

* fix: add timeouts to network prover calls to prevent deadlock

* refactor: improve

* chore: set NETWORK_CALL_TIMEOUT_SECS to 15s

fakedev9999 committed:
refactor: use cached dispute state for challenger (#629)

* fix(fault-proof): fix proposer with canonical chain tracking

* refactor: improve game resolution

* refactor: improve game credit claim

* refactor: improve game defense

* chore: add comments

* chore: rename to update_canonical_head_if_further

* chore: rename to is_XXX for boolean returning func

* refactor: remove early return from `refresh_state`

* fix

* fix

* refactor: refactor fix-proposer PR (#627)

* move run() to the top

* move state init from refresh state to proposer init

* huge refac

* remove is_finalized

* remove update_canonical_head_if_further

* move below new

* switch orders

* compute canonical head

* remove state snapshot

* typo

* refac remove subtree

* fmt

* update fetch_game

* rename to sync_state

* remove is_descendant_of_anchor

* refac remove_subtree

* more refac (#628)

* wip

* remove for claim_bond_for_game

* fix

* clippy

* fix

* fix

* fix

* clippy

* fix

* refac

* fix: drop if parent game is not in cache

* switch orders

* fix is_parent_resolved

* update testing doc

* remove cursor and set_cursor

* improve fetch_game

* improve sync_games

* misspell

* refac: cursor+deadline scan with gameType filter

- Remove MAX_GAMES_TO_CHECK_FOR_CHALLENGE
- Simplify malicious challenge

* move run to top

* wip

* huge refac + compiles

* more refac

* more refac

* more refac

* polish docs

* more refac

* udpate comment

* more refac

* update comments

* fix compiler error

* fix

fakedev9999 committed:
fix: validate cached checkpoints against contract before reuse (#641)

* fix: validate cached checkpoints against contract before reuse

* forge fmt for foundry 1.4.1-stable

fakedev9999 committed:
feat: add json logging (#643)

* feat: add JSON log format option to logger setup

* feat: add ansi feature flag and respect NO_COLOR in logger

* fix: clean up imports and formatting in logger.rs

* chore: clippy

---------

Co-authored-by: Victor Castell <0x@vcastellm.xyz>

fyInALT committed:
feat: add --slow for deploy to testnet or mainnet (#599)

Co-authored-by: Aurélien <3535019+leruaa@users.noreply.github.com>

leruaa committed:
feat: add support for GCP HSM (#638)

* feat: add support for GCP HSM

* feat: use alloy-signer-gcp

* chore: add signer methods

* fix: signer test

* fix: typo

Farhad-Shabani committed:
feat: optional `L1_BEACON_RPC` (#645)

* feat: optional l1 beacon url

* nit

* refactor: return Result instead of Option

* fix: L1_BEACON_RPC in book

* nit

* fix: make fmt happy

fakedev9999 committed:
refactor: refactor run_range_program (#635)

* refactor: refactor run_range_program

* fix

* fmt

* chore: bump celestia range elf

* chore: bump elfs

fakedev9999 committed:
fix: return false for parent game challenger wins check when genesis (#650)

* fix: return false for parent game challenger wins check when genesis

* fix

---------

Co-authored-by: seolaoh <osa8361@gmail.com>

leruaa committed:
feat: add a timeout for network calls (#646)

piersy committed:
Merge remote-tracking branch 'upstream/main' into piersy/merge-upstream-fusaka-readiness

Plain merge, no conflicts addressed

 Conflicts:
	.github/workflows/ci.yaml
	.github/workflows/cycle-count-diff-eigenda.yml
	.github/workflows/e2e-tests.yml
	.github/workflows/integration-tests.yml
	Cargo.lock
	Cargo.toml
	README.md
	book/advanced/node-setup.md
	book/advanced/verify-binaries.md
	book/fault_proofs/experimental/celestia.md
	book/fault_proofs/experimental/eigenda.md
	book/fault_proofs/proposer.md
	book/fault_proofs/testing.md
	book/validity/contracts/update-parameters.md
	book/validity/experimental/eigenda.md
	contracts/script/fp/DeployOPSuccinctFDG.s.sol
	elf/aggregation-elf
	elf/eigenda-range-elf-embedded
	elf/range-elf-bump
	elf/range-elf-embedded
	fault-proof/Cargo.toml
	fault-proof/bin/challenger.rs
	fault-proof/bin/proposer.rs
	fault-proof/src/challenger.rs
	fault-proof/src/config.rs
	fault-proof/src/lib.rs
	fault-proof/src/proposer.rs
	fault-proof/tests/common/anvil.rs
	fault-proof/tests/common/env.rs
	fault-proof/tests/common/process.rs
	fault-proof/tests/e2e.rs
	justfile
	programs/range/eigenda/Cargo.toml
	programs/range/eigenda/src/main.rs
	programs/range/utils/src/lib.rs
	scripts/prove/Cargo.toml
	scripts/utils/bin/fetch_and_save_proof.rs
	utils/build/src/lib.rs
	utils/celestia/client/src/executor.rs
	utils/client/Cargo.toml
	utils/client/src/client.rs
	utils/client/src/precompiles/mod.rs
	utils/client/src/witness/executor.rs
	utils/eigenda/client/Cargo.toml
	utils/eigenda/client/src/executor.rs
	utils/eigenda/host/Cargo.toml
	utils/eigenda/host/src/host.rs
	utils/eigenda/host/src/witness_generator.rs
	utils/ethereum/client/src/executor.rs
	utils/host/Cargo.toml
	utils/host/src/witness_generation/traits.rs
	validity/src/proof_requester.rs
	validity/src/proposer.rs

piersy committed:
Resolve Cargo.toml conflict

Somehow git had duplicated the workspace.dependencies, so this
de-duplicates it as well as resolving the conflicts.

piersy committed:
Resolve conflicts under fault-proof

piersy committed:
Resolve conflicts in DeployOPSuccinctFDG.s.sol

piersy committed:
Resolve conflicts in justfile

piersy committed:
Resolve conflicts under programs/range

piersy committed:
Resolve conflicts under scripts

piersy committed:
Resolve conflicts under validity

Just accepted upstream since we had no prior diff on these files from
our previous "merge base" (we didn't have a real merge base since we had
not merged)

fakedev9999 committed:
chore: bump kona to kona-client/v1.1.4

piersy committed:
Resolve conflicts under utils

piersy committed:
Cargo.toml format

piersy committed:
Cargo.toml use celo fork of kona at v1.1.3

piersy committed:
Fix newly added function args in OPSuccinctDataFetcher

Use CeloRollupConfig instead of RollupConfig

piersy committed:
Allow unused imports, helps to reduce upstream diff

piersy committed:
Resolve conflicst in Cargo.lock

piersy committed:
Fixup justfile

piersy committed:
Update elfs

piersy committed:
Fixup conflict resolution  utils/client/src/witness/executor.rs

fakedev9999 committed:
fix

Farhad-Shabani committed:
fix: re-enable CI tests + resolve Anvil conflict + forge fmt (#660)

* fix: drop Anvil if test panics

* fix: validity docstring tests

* fix: forge fmt

* fix: ci.yaml + test-threads=1

* fix: forge fmt with nightly

* refactor: e2e feature flag + run from Justfile

* fix: bring back L1 RPCs for now

* nit: happy fmt

* fix: testing guide cmd

* fix: not skip test_honest_proposer/challenger

fakedev9999 committed:
wip

fakedev9999 committed:
fix

fakedev9999 committed:
Merge branch 'main' into taehoon/bump-kona

piersy committed:
Resolve conflicts under .github

piersy committed:
Resolve conflicts in README.md

piersy committed:
Resolve conflicts under book

piersy committed:
Update elfs

piersy committed:
Update celo-kona to latest v1.1.3 kona version

fakedev9999 committed:
chore: bump hana

fakedev9999 committed:
chore: bump hokulea

piersy committed:
Update Cargo.lock

Looks like something could be up with hokulea, we have multiple versions
appearing

piersy committed:
Merge commit 'e53406c9' into piersy/merge-upstream-fusaka-readiness

Plain merge, no conflict resolution

 Conflicts:
	.github/workflows/ci.yaml
	Cargo.lock
	elf/aggregation-elf
	elf/eigenda-range-elf-embedded
	elf/range-elf-bump
	elf/range-elf-embedded
	justfile
	resources/g1.point

piersy committed:
Manually resolve conflicts

piersy committed:
Resolve conflicts in Cargo.lock

piersy committed:
Update celo-kona dep to version supporting kona v1.1.6

piersy committed:
Update use patched kona v1.1.6 version

fakedev9999 committed:
chore: bump hokulea

piersy commented on utils/host/src/fetcher.rs:
```diff
@@ -314,6 +332,36 @@ impl OPSuccinctDataFetcher {
     }
 
+    /// Fetch and save the L1 config based on the rollup config's L1 chain ID.
+    async fn fetch_and_save_l1_config(rollup_config: &CeloRollupConfig) -> Result<PathBuf> {
```
Needed to update this to be compatible with celo types

piersy commented on fault-proof/bin/proposer.rs:
```diff
@@ -29,14 +29,18 @@ struct Args {
+    // TODO: check with seola/yoshiki about this line, I can't see that it ever exsited in upstream.
+    // seems it was added in https://github.com/celo-org/op-succinct/commit/2a5653d1381a8db2e19c042325c8bcbe43d55a68
+    // see this thread https://clabsco.slack.com/archives/C08C4523K2R/p1750410628172889
     rustls::crypto::ring::default_provider().install_default().unwrap();
```
Hey @seolaoh & @Kourin1996 could you let me know if you think we still need this rustls call?

seolaoh replied:
Yes, I believe we still need this. We could try removing it and testing whether proving works without this line, but at the time it was added, we were the only ones encountering an issue without this line, upstream didn't have an issue.

seolaoh replied:
Can we remove the TODO line in the comment?

piersy commented on contracts/script/fp/DeployOPSuccinctFDG.s.sol:
```diff
@@ -58,15 +58,15 @@ contract DeployOPSuccinctFDG is Script, Utils {
         FDGConfig memory config = readFDGJson("opsuccinctfdgconfig.json");
 
         // Deploy contracts
-        DeployedContracts memory contracts = deployContracts(config);
```
@Mc01, would appreciate your 👀 on this file!

piersy committed:
Merge tag 'v3.3.1-rc.3' into piersy/merge-upstream-fusaka-readiness

Plain merge, no conflict resolution

 Conflicts:
	Cargo.lock
	elf/eigenda-range-elf-embedded

piersy committed:
Resolve conflicts in Cargo.lock

piersy committed:
Ignore clippy error

piersy committed:
Update elfs

seolaoh commented on fault-proof/src/proposer.rs:
```diff
@@ -218,9 +564,14 @@ where
                 .prove(&self.prover.range_pk, &sp1_stdin)
                 .compressed()
-                .strategy(FulfillmentStrategy::Reserved)
                 .skip_simulation(false)
```
Let's follow upstream as they introduced `self.config.range_cycle_limit` so this `skip_simulation(false)` doesn't work anymore. If we really want to force the proposer to simulate proof generation and estimate `cycle_limit` and `gas_limit`, we should remove `.cycle_limit(self.config.range_cycle_limit)` and `.gas_limit(self.config.range_gas_limit)`. But I think it's enough to have such large default value for `cycle_limit` and `gas_limit` and skip simulation.

```suggestion
                .skip_simulation(true)
```

piersy replied:
Hey @seolaoh, so previously did we do the simulation to avoid the chance of us using far too many cycles on the proving network?

seolaoh replied:
Previously the `cycle_limit` was set as a constant which was much smaller than current default value of `self.config.range_cycle_limit`. So we need to manually change the value when we run into out of cycle_limit error. So we changed it to simulate the execution and estimate the `cycle_limit` dynamically. But it takes quite a long time to simulate that and now we have a much larger value for the config, so I think it's fine to remove the simulation step.

seolaoh commented on fault-proof/src/proposer.rs:
Though this proposer can't handle game type transition, I'll create a following PR based on [this branch](https://github.com/celo-org/op-succinct/commits/seolaoh/fix-refactored-proposer/) to have a workaround for game type transition.

seolaoh commented on book/advanced/verify-binaries.md:
```diff
@@ -38,18 +38,18 @@ To build the binaries, run:
 cd programs/range/ethereum
-cargo prove build --output-directory ../../../elf --elf-name range-elf-bump --docker --tag v5.2.1
-cargo prove build --output-directory ../../../elf --elf-name range-elf-embedded --docker --tag v5.2.1 --features embedded
+cargo prove build --output-directory ../../../elf --elf-name range-elf-bump --docker --tag v5.2.2
```
Could you update the tag in `.github/workflows/elf.yml` to v5.2.2 as well?

seolaoh commented on utils/host/src/fetcher.rs:
```diff
@@ -50,8 +52,7 @@ impl Default for OPSuccinctDataFetcher {
     pub l1_rpc: Url,
-    // TODO(fakedev9999): Make optional if possible.
-    pub l1_beacon_rpc: Url,
+    pub l1_beacon_rpc: Option<Url>,
```
@ezdac L1_BEACON_URL becomes optional!

seolaoh commented on utils/signer/src/lib.rs:
```diff
@@ -21,13 +26,16 @@ pub enum Signer {
     /// The local signer.
     LocalSigner(PrivateKeySigner),
+    /// Cloud HSM signer using Google.
+    CloudHsmSigner(GcpSigner),
```
@ezdac Also something for HSM is added

seolaoh commented on .github/workflows/cycle-count-diff-celestia.yml:
```diff
@@ -0,0 +1,88 @@
+on:
+  pull_request:
+    paths:
+      - 'elf/celestia-range-elf-embedded'
```
Let's disable this
```suggestion
  workflow_dispatch:
  # pull_request:
  #   paths:
  #     - 'elf/celestia-range-elf-embedded'
```

Mc01 commented on contracts/src/validity/OPSuccinctL2OutputOracle.sol:
```diff
@@ -500,6 +496,16 @@ contract OPSuccinctL2OutputOracle is Initializable, ISemver {
+    /// @param _l1BlockNumber The L1 block number of the output to propose.
+    /// @param _proof The proof of the output to propose.
+    /// @param _proverAddress The address of the prover to use.
+    /// @return _game The dispute game created.
```
Duplicate - to delete
```suggestion
```

Mc01 reviewed (commented):
Nothing serious.
Just formatting & one change which looks like a merging error

Mc01 commented on contracts/script/fp/DeployOPSuccinctFDG.s.sol:
We might want back those lines:
```
// Log deployed addresses.
console.log("Factory Proxy:", address(factoryProxy));
console.log("Game Implementation:", address(gameImpl));
console.log("SP1 Verifier:", sp1Config.verifierAddress);
```
from https://github.com/celo-org/op-succinct/pull/41/commits/64d60c25ab6d4bcd251b3855a4e074071ef1ee38

piersy committed:
Bump hokulea to rev 90404fb (fix use of bpos)

This revision comes from the upstream bump--sp1-cc-with-bpo-temp branch
which pulls in a fixed version of the rsp crate. See
https://github.com/succinctlabs/rsp/pull/159.

Also bumps celo-kona to a version that uses hokulea rev 90404fb.

piersy committed:
Update elfs and update CI elf check to use v5.2.2

piersy committed:
Update comment

piersy committed:
Disable CI cycle count diff celestia workflow

Co-authored-by: Seola Oh <osa8361@gmail.com>

piersy committed:
Remove duplicated comment

piersy committed:
Add logging back to DeployOPSuccinctFDG.s.sol

piersy committed:
Skip proposer execution simulation

Upstream increased the cycle limit to a point where we don't need to worry about running into the limit, and we can therefore skip the simulation.

Co-authored-by: Seola Oh <osa8361@gmail.com>

piersy commented:
Hey @seolaoh I think this is pretty much complete now.

seolaoh reviewed (approved):
LGTM!
